### PR TITLE
[WIP]Makeshift mod and bayonet update.

### PR DESCRIPTION
**Pull request by Noctifer-de-Mortem**
Made some changes to the makeshift mod arround bayonets and a couple of changes in core that were required to get the desired effect. Before the "standard" bayonet could be use with smgs and crossbows, restored that once more. I also update the modinfo for the makeshift mod to reflect its intent more. Finally I update some recipes to make makeshift bayonets easier to make, as of their intent.

Changelog in commit descriptions.

### CODE REVIEW

File: data/mods/makeshift/recipe.json [hunk 1]
```diff
@@ -93,5 +93,65 @@
...
+      [ "estoc", 1 ],
+      [ "zweihander", 1 ],
+      [ "nodachi", 1 ],
+      [ "wakizashi", 1 ],
+      [ "scythe_war", 1 ],
+      [ "scythe", 1 ],
+      [ "glaive", 1 ],
+      [ "naginata", 1 ],
+      [ "qiang", 1 ],
+      [ "lajatang", 1 ],
```
Comment (Coolthulhu) [inline data/mods/makeshift/recipe.json]: Tabs

**Comment by Noctifer-de-Mortem** [comment]
After testing the changes on the game, they seem to work. Is it a lint/astyle issue?

**Comment by kolsurma** [comment]
from http://ci.narc.ro/job/Cataclysm-Jently/71312//consoleText
```
Linting data/mods/makeshift/recipe.json
ERROR: Unmatched contex 'recipe<recipe>:override'
make: *** [obj/lint.cache] Error 65
```
The override flag doesn't seem to exist for recipes.

**Comment by Noctifer-de-Mortem** [comment]
Ah, well then, will have to make a few changes then.

**Comment by Noctifer-de-Mortem** [comment]
Woo! Got it to work!

**Comment by Noctifer-de-Mortem** [comment]
I am going to close this PR and open a new one since it is giving me new linting errors that I can't seem to solve.

**Pull request closed without merging**